File: 04_algo_book/java/src/baek/B2164.java
package baek;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;

/**
 * 카드2
 */
public class B2164 {

    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(bf.readLine());

        Queue<Integer> Q = new LinkedList<>();

        for (int i = 1; i <= n; i++) {
            Q.offer(i);
        }

        while (Q.size() > 1) {
            Q.poll();
            Integer poll = Q.poll();
            Q.offer(poll);
        }

        System.out.println(Q.poll());
    }
}

File: 00_algo_book/java/src/baek/B2164.kt
package baek

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.LinkedList

fun main() {

    val bf = BufferedReader(InputStreamReader(System.`in`))

    val n = bf.readLine().toInt()
    val Q = LinkedList<Int>()

    for (i in 1..n) {
        Q.offer(i)
    }


    while (Q.size > 1) {
        Q.poll()
        Q.offer(Q.poll())
    }

    println(Q.poll())
}